src/machine-learning/machine-learning.ts

```diff
@@ -246,7 +248,11 @@ export class Model {
 
   public readonly tfliteModel?: TFLiteModel;
 
-  constructor(model: ModelResponse) {
+  private readonly activeOperations?: OperationResponse[];
+  private readonly app?: FirebaseApp;
```

Comment:
This is a bad idea. `FirebaseApp` is a complex and heavy weight object, and attaching that onto things that are returned to the user is not going to work out well. I'd be more comfortable, if this just kept a reference to the API client instance instead. I have reservations about this approach too, but I don't see any other way around it.

Also now that this class has private fields and functions, let's also define a `toJSON()` method in case somebody tries to serialize a model object into JSON.

Reply:
Done. (Sorry, I thought client was bigger than app)